tchannel/response.py

```diff
@@ -74,17 +75,31 @@ class TransportHeaders(object):
         'scheme',
     )
 
-    def __init__(self,
-                 failure_domain=None,
-                 scheme=None,
-                 **kwargs):
```

Comment:
This a risky part in python. We can't prevent others using ``TransportHeaders`` directly. Therefore though we can make sure we pass valid arguments, we can't guarantee the behavior that whoever try to instantiate the ``TransportHeaders`` directly.